Cython/Compiler/Nodes.py

```diff
@@ -1811,7 +1811,10 @@ def generate_function_definitions(self, env, code):
         self.generate_keyword_list(code)
 
         # ----- GIL acquisition
-        acquire_gil = self.acquire_gil
+        if self.entry.is_cgetter:
+            acquire_gil = False
+        else:
+            acquire_gil = self.acquire_gil
```

Comment:
The getter function might still be defined with a `with gil` declaration. Is there anything specific that didn't work if you leave this out?